ui/shared/components/panel/LoadWorkspaceDataForm.jsx

```diff
@@ -104,16 +104,37 @@ const createProjectFromWorkspace = postWorkspaceValues(
   window.location.href = `/project/${responseJson.projectGuid}/project_page`
 })
 
-const FORM_WIZARD_PAGES = [
-  { fields: [AGREE_CHECKBOX], onPageSubmit: postWorkspaceValues('grant_access') },
-  { fields: [DATA_BUCK_FIELD, SAMPLE_TYPE_FIELD, REQUIRED_GENOME_FIELD], onPageSubmit: postWorkspaceValues('validate_vcf') },
+const addDataFromWorkspace = projectGuid => (values, onSuccess) => (
+  new HttpRequestHelper(`/api/project/${projectGuid}/add_workspace_data`, (responseJson) => {
+    if (responseJson.success) {
+      onSuccess()
+    }
+  }).post({ ...values, uploadedFileId: values.uploadedFile?.uploadedFileId }))
```

Comment:
this shares too much logic with `postWorkspaceValues`/`createProjectFromWorkspace` to warrant being a separate function. Just modify `postWorkspaceValues` to take an optional third argument of `formatUrl` and then change the url logic to 
```
formatUrl ? formatUrl(values) : `/api/create_project_from_workspace/${workspaceNamespace}/${workspaceName}/${path}`
```
Then make a shared helper for the 2 submit functions:
```
const postSubmitValues =  formatUrl => postWorkspaceValues(
  'submit', ({ uploadedFile, ...values }) => ({ ...values, uploadedFileId: uploadedFile.uploadedFileId }), formatUrl,
)
const createProjectFromWorkspace = postSubmitValues()((responseJson) => {
  window.location.href = `/project/${responseJson.projectGuid}/project_page`
})
const addDataFromWorkspace = postSubmitValues(({ ...projectGuid }) => `/api/project/${projectGuid}/add_workspace_data`)
```

Reply:
okay, but we don't want to delay form closing. The modal should never automatically close. When the form is successfully submitted it just changes the content from a form to a success message and the modal stays open. I don't see a good reason to add all the complexity of connecting this form to the modal when it should never interact with the modal

Reply:
Okay, I removed the modal. For creating a new project, it will jump to the project page after submission. For adding new data, the modal keeps opening after displaying the success message until the user clicks on the `x` of the modal or anywhere outside of the modal.